### PR TITLE
Microbit events transform and added using Boolean Values

### PR DESCRIPTION
**Pull request by K00ry**
Rather than using **when** and **wait** for the Microbit Temperature, Buton A/B pressed, and pins pressed Events, added **is** boolean blocks so they could be used with **do while** loops to create events programmatically.

### CODE REVIEW

**Review by juniovitorino — COMMENTED** [review]
@K00ry Overall, we can recover this commented code if necessary using git. I advocate we stop commenting out code and remove all on this and all our other projects. What would you think about getting rid of this commented code?

**Comment by K00ry** [comment]
@juniovitorino I checked again; I don't see any commented code in this PR; I think you were mistaken PXT decorators with a regular commented-out code.

File: libs/core/sims.d.ts [hunk 1]
```diff
@@ -188,12 +188,26 @@ declare namespace Microbit {
...
     //% variable.defl="Microbit 1"
-    //% group="Events"
-    //% shim=Microbit::whenButtonPressed
-    function whenButtonPressed(variable: BBCMicrobit, buttonOption: MicrobitButtonOptions, velocityOption: MicrobitButtonVelocity): void;
+    //% group="Values"
+    //% shim=Microbit::isMicrobitPinPressed
+    function isMicrobitPinPressed(variable: BBCMicrobit, analogPinOption: MicrobitAnalogPinOptions): boolean;
+
+    //% blockId="microbit_temperature_changed" block="is %variable temperature changed"
+    //% variable.shadow=variables_get
```
Comment (K00ry) [inline libs/core/sims.d.ts:199]: @juniovitorino these are PXT decorators.

Comment (juniovitorino) [reply libs/core/sims.d.ts:199]: Hahaha, ok 😅 my apologies.

**Pull request merged**